pydantic/env_settings.py

```diff
@@ -232,7 +235,16 @@ def explode_env_vars(self, field: ModelField, env_vars: Mapping[str, Optional[st
             env_var = result
             for key in keys:
                 env_var = env_var.setdefault(key, {})
-            env_var[last_key] = env_val
+
+            if env_val:
+                try:
```

Comment:
The problem here is we're calling `json.loads` on every field which:
1. could have some weird side effects - e.g. you want `"foobar"` but `json_loads` converts it to `foobar`
2. It could effect performance

We could:
* not do this and document it?
* check if `env_val` starts with `{` or `[` and only parse json in that case